[boost] Update to 1.72.0

yurybura opened the pull request:
Update Boost libarary to 1.72.0:
*  `generate-ports.ps1` script was modified to support new version and was run to re-generate related ports;
* `boost-python` was fixed to find Python3 even Python2 port is installed.
 
Fixes issue #9298 #6310

yurybura committed:
[boost-python] Support Python2 as feature, fix build with installed Python2

yurybura committed:
[boost-python] Update generate script to support Python2 mod

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
Re-generate boost ports

yurybura committed:
[boost-modular-build-helper] Update to 1.71.
[boost-build] Update to 1.71.

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
[boost-modular-build-helper] Remove manual library name changing
[boost-iostreams] Use boost-modular-headers.cmake script

yurybura committed:
Delete temporary file.

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
Revert changes in modular build script.

yurybura committed:
Update ports versions.

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
[boost-modular-build-helper] Fix build boost-iostreams with LZMA and ZSTD

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
[boost] Update to 1.72.0

cenit commented on ports/boost-python/portfile.cmake:
```diff
@@ -5,13 +5,14 @@ include(vcpkg_common_functions)
 
 # Find Python. Can't use find_package here, but we already know where everything is
-file(GLOB PYTHON_INCLUDE_PATH "${CURRENT_INSTALLED_DIR}/include/python[0-9.]*")
+set(PYTHON_DIR_NAME "python3\.*")
```
This dir_name won't work on *nix. Please change the backslash to a forward slash

yurybura replied:
`\.` matches the character `.` literally. Include path must be [...]/python3.6 or [...]/python3.7, etc.

yurybura replied:
Fixed.

cenit commented on scripts/boost/post-source-stubs/python.cmake:
```diff
@@ -1,5 +1,6 @@
 # Find Python. Can't use find_package here, but we already know where everything is
-file(GLOB PYTHON_INCLUDE_PATH "${CURRENT_INSTALLED_DIR}/include/python[0-9.]*")
+set(PYTHON_DIR_NAME "python3\.*")
```
Same here

yurybura replied:
Fixed.

cenit reviewed (commented):
Please also edit the generate-ports.ps1 to remove `include(vcpkg_common_functions)`

yurybura committed:
[libtorrent] Fix build with Boost 1.72.0
Original PR: https://github.com/arvidn/libtorrent/pull/4091

yurybura committed:
Update ci.baseline.txt:
- fail libgo:* due to use the removed class from Boost.Context 1.72.0 (reported issue https://github.com/yyzybb537/libgo/issues/214)
- pass boost-context:arm64-windows
- fix typo

yurybura committed:
Folly only supports the x64 architecture

yurybura committed:
fail boost-coroutine:arm64-windows=fail and boost-fiber:arm64-windows=fail because they use fcontext_t from Boost.Context which is not implemented for Windows on ARM64 (reported issue https://github.com/boostorg/context/issues/130)

yurybura committed:
Fix globbing-expressions

yurybura commented:
> This means that libgo is not buildable anymore in any os nor any config?

Temporary yes, this library is not compatible with Boost.Context 1.72.0.

PhoebeHui reviewed (approved):
Thanks for the PR!

ras0219-msft commented on scripts/boost/generate-ports.ps1:
```diff
@@ -164,9 +154,6 @@ function Generate()
                 ")"
             )
         }
-        elseif ($Name -eq "iostreams")
```
Please revert the changes to how `boost-iostreams` is handled and submit it as a separate PR

yurybura replied:
Reverted.

PhoebeHui commented:
Related to https://github.com/microsoft/vcpkg/pull/8372

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
Add vcpkg_buildpath_length_warning(37) for boost-system.
Revert boost-iostreams port changes.

yurybura commented on ports/boost-system/portfile.cmake:
```diff
@@ -2,13 +2,11 @@
 
 include(vcpkg_common_functions)
 
-vcpkg_buildpath_length_warning(37)
```
I've added it for boost-system only (like it was before).

JackBoosY commented:
/azp run

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

JackBoosY commented:
/azp run

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

dan-shaw commented:
/azp run

yurybura commented:
Dear @dan-shaw, CI was failed with empty log,.. could you help to investigate the reason?

JackBoosY commented:
/azp run

dan-shaw commented:
/azp run

dan-shaw commented:
Should work now

ras0219-msft commented:
/azp run

ras0219-msft commented:
/azp run

JackBoosY commented:
/azp run

yurybura committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg

yurybura committed:
Update ci.baseline.txt

yurybura commented:
Finally all checks have done. Please merge or request changes. Thanks a lot.

vicroms commented:
Thanks for the PR!
